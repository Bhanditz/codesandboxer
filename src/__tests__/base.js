import cases from 'jest-in-case';
import {
  getAllImports,
  replaceImport,
  replaceImports,
  parseDeps,
  parseFile,
  getCSBData,
} from '../index';

const codeImportTests = [
  { name: 'simple import', code: `import a from 'b'` },
  { name: 'spread import', code: `import { a } from 'b'` },
  { name: 'two imports', code: `import a from 'b' import c from 'd'` },
  {
    name: 'multiline imports',
    code: `import a from 'b'
import c from 'd'`,
  },
  { name: 'two spread imports', code: `import { a, b } from 'c'` },
  {
    name: 'two spread imports multiline',
    code: `import {
  a,
  b
} from 'c'`,
  },
  { name: 'no spaces', code: `import {a} from 'b'` },
  {
    name: 'dev and peer deps',
    code: `import {a} from 't' import s from 'z' import y from 'x'`,
  },
  {
    name: 'relativeImport',
    code: `import {a} from './c'`,
  },
  {
    name: 'using regex pattern',
    code: `import a from './c/somewhere' import b from './c/anywhere'`,
    old: './c/*',
    new: 'c/',
  },
];

cases(
  'getAllImports()',
  ({ code }) => {
    let mpts = getAllImports(code);
    console.log(mpts);
    expect(mpts).toMatchSnapshot();
  },
  codeImportTests,
);

cases(
  'replaceImport()',
  ({ code, old = 'b', newImp = 'zzz' }) => {
    let newCode = replaceImport(code, old, newImp);
    expect(newCode).toMatchSnapshot();
  },
  codeImportTests,
);

cases(
  'replaceImports()',
  ({ files, replaces }) => {
    let code = replaceImports(files, replaces);
    expect(code).toMatchSnapshot();
  },
  [
    {
      name: 'replace imports',
      files: `import a from 'b' import c from 'd'`,
      replaces: [['b', 'z'], ['d', 'arg']],
    },
  ],
);

const fakePKGJSON = {
  name: 'not-a-real-name',
  version: '0.0.1',
  devDependencies: { b: 'v1.0.0' },
  peerDependencies: { z: 'v1.0.0', x: 'v0.3.0' },
  dependencies: { x: 'v1.0.0' },
};

cases(
  'parseDeps()',
  ({ code }) => {
    let imports = getAllImports(code);
    let parsedImports = parseDeps(fakePKGJSON, imports);
    // console.log(imports);
    // console.log(parsedImports);
    expect(parsedImports).toMatchSnapshot();
  },
  codeImportTests,
);

cases(
  'ParseFile()',
  async ({ file, pkgJSON }) => {
    let parsedFile = await parseFile(file, pkgJSON);
    expect(parsedFile).toMatchSnapshot();
  },
  [
    {
      name: 'simple parse',
      file: `import a from 'b'; import c from './c'`,
      pkgJSON: fakePKGJSON,
    },
    {
      name: 'simple parse file promise',
      file: Promise.resolve(`import a from 'b'; import c from './c'`),
      pkgJSON: fakePKGJSON,
    },
    {
      name: 'simple parse pkgJSON promise',
      file: `import a from 'b'; import c from './c'`,
      pkgJSON: Promise.resolve(fakePKGJSON),
    },
  ],
);

cases(
  'getCSBData()',
  async ({ file, pkgJSON, config }) => {
    let csbData = await getCSBData(file, pkgJSON, config);
    expect(csbData).toMatchSnapshot();
  },
  [
    {
      name: 'simple parse',
      file: `import a from 'b'; import c from './c'`,
      pkgJSON: fakePKGJSON,
      config: {},
    },
    {
      name: 'adding new dependency',
      file: `import a from 'b'; import c from './c'`,
      pkgJSON: fakePKGJSON,
      config: { providedDeps: { d: '1.1.0' } },
    },
    {
      name: 'adding extra file',
      file: `import a from 'b'; import c from './lib/c'`,
      pkgJSON: fakePKGJSON,
      config: {
        providedFiles: { 'lib/c': { content: "export default 'something'" } },
      },
    },
  ],
);
