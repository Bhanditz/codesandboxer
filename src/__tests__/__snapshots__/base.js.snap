// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParseFile() simple parse 1`] = `
Object {
  "deps": Object {
    "b": "v1.0.0",
  },
  "file": "import a from 'b'; import c from './c'",
  "internalImports": Array [
    Array [
      "import c from './c'",
      "./c",
    ],
  ],
}
`;

exports[`ParseFile() simple parse file promise 1`] = `
Object {
  "deps": Object {
    "b": "v1.0.0",
  },
  "file": "import a from 'b'; import c from './c'",
  "internalImports": Array [
    Array [
      "import c from './c'",
      "./c",
    ],
  ],
}
`;

exports[`ParseFile() simple parse pkgJSON promise 1`] = `
Object {
  "deps": Object {
    "b": "v1.0.0",
  },
  "file": "import a from 'b'; import c from './c'",
  "internalImports": Array [
    Array [
      "import c from './c'",
      "./c",
    ],
  ],
}
`;

exports[`getAllImports() dev and peer deps 1`] = `
Array [
  Array [
    "import {a} from 't'",
    "t",
  ],
  Array [
    "import s from 'z'",
    "z",
  ],
  Array [
    "import y from 'x'",
    "x",
  ],
]
`;

exports[`getAllImports() multiline imports 1`] = `
Array [
  Array [
    "import a from 'b'",
    "b",
  ],
  Array [
    "import c from 'd'",
    "d",
  ],
]
`;

exports[`getAllImports() no spaces 1`] = `
Array [
  Array [
    "import {a} from 'b'",
    "b",
  ],
]
`;

exports[`getAllImports() relativeImport 1`] = `
Array [
  Array [
    "import {a} from './c'",
    "./c",
  ],
]
`;

exports[`getAllImports() simple import 1`] = `
Array [
  Array [
    "import a from 'b'",
    "b",
  ],
]
`;

exports[`getAllImports() spread import 1`] = `
Array [
  Array [
    "import { a } from 'b'",
    "b",
  ],
]
`;

exports[`getAllImports() two imports 1`] = `
Array [
  Array [
    "import a from 'b'",
    "b",
  ],
  Array [
    "import c from 'd'",
    "d",
  ],
]
`;

exports[`getAllImports() two spread imports 1`] = `
Array [
  Array [
    "import { a, b } from 'c'",
    "c",
  ],
]
`;

exports[`getAllImports() two spread imports multiline 1`] = `
Array [
  Array [
    "import {
  a,
  b
} from 'c'",
    "c",
  ],
]
`;

exports[`getAllImports() using regex pattern 1`] = `
Array [
  Array [
    "import a from './c/somewhere'",
    "./c/somewhere",
  ],
  Array [
    "import b from './c/anywhere'",
    "./c/anywhere",
  ],
]
`;

exports[`getCSBData() adding extra file 1`] = `
Object {
  "dependencies": Object {
    "b": "v1.0.0",
    "not-a-real-name": "0.0.1",
    "react": "latest",
    "react-dom": "latest",
  },
  "files": Object {
    "example.js": Object {
      "content": "import a from 'b'; import c from './lib/c'",
    },
    "index.html": Object {
      "content": "<div id=\\"root\\"></div>",
    },
    "index.js": Object {
      "content": "/**
  This CodeSandbox has been automatically generated using a cool tool.

  This generator does not follow relative imports beyond those that reference the
  module root, and as such, other relative imports may fail to load.

  If you are experiencing difficulty contact @noviny on twitter
*/
import React from 'react';
import ReactDOM from 'react-dom';
import Example from './example';

ReactDOM.render(
<Example />,
document.getElementById('root')
);",
    },
    "lib/c": Object {
      "content": "export default 'something'",
    },
    "package.json": Object {
      "content": "{
  \\"name\\": \\"simple-example\\",
  \\"version\\": \\"0.0.0\\",
  \\"description\\": \\"A simple example deployed to CodeSandbox\\",
  \\"main\\": \\"index.js\\",
  \\"dependencies\\": {\\"b\\":\\"v1.0.0\\",\\"not-a-real-name\\":\\"0.0.1\\",\\"react\\":\\"latest\\",\\"react-dom\\":\\"latest\\"}
}",
    },
  },
  "parameters": "N4IgZglgNgpgziAXKCA7AJjAHgOgBYAuAtlEqAMYD2qBMNSIAPOhAG4AEE6AvADogAnSpQL8AfIwD0LVmJABfADQg0mXACsEyEFRp0CDSQCojvVO3YAVPBDjsAwpUwBlAIYYARpSzs8rux4wdOyuAK4ElESuBBDkrlBQAJ7sAOZ0MALRMOjsoXBoKSHsVJRQ7BGlOGZmFta2qemZEQLs6JTw7Kgi7GClUJQA7uwCMFDRbDCcRAAOlAIEATCJ1DkEeJRwk2vRwzBgGXTkW3gwNexETqGww8IEiiEYIXZwoeR49yInLSNjMayTEBmcwW51cyTArmg5Uo7H6rnQVVQZwAkmB2MtQiERuxsNMMhBDgVWhAwJByFcCMldARXOQCOwAAJdVhoZLUcoDCAEWgCMxGSRmQGzebsABKMFp9LAQiI7AA5CNJXKANyCoEi8WSgAiAHkALI9GXyxV0gC0bSIKrVwvpAFEsK4ZtdpZF5ThJNhHdNYFakahNXTdXqcCMMBkABRmRj2r3XSRiRRmNrkoj6HBpAi22CpmgAIUSyPQ4YVtzlAEozGXlQplJ6nTAcJoyDpqLR6IgVOr6a5Da65R4VVMbcVe7K5e6oBAPJJyHKayBprSANauNKNuDUZvU_QMYBnfioR2nJDsfj5eumuve4-J8ynkD_AT5aj8RD3gAMOE_7_4t4s_EwOByAECBphiF8T34ABBdhz2vHEHXrVoYG9ShEmyaEHCcGA3E8bxf33EAojQV971UbB1wIu8AJQuhMFQcgCTgUi9xADxX34VgAEYvy_AiQC6AhTVcU1FSgU1D1TDiQG_HAuP4k1RCQfhfngJTE0ECUzQtaTVLgJT5DMeR50nadyC3Vsdw7XFgWQiEKXlDdUzWAo53kdygA",
}
`;

exports[`getCSBData() adding new dependency 1`] = `
Object {
  "dependencies": Object {
    "b": "v1.0.0",
    "d": "1.1.0",
    "not-a-real-name": "0.0.1",
    "react": "latest",
    "react-dom": "latest",
  },
  "files": Object {
    "example.js": Object {
      "content": "import a from 'b'; import c from './c'",
    },
    "index.html": Object {
      "content": "<div id=\\"root\\"></div>",
    },
    "index.js": Object {
      "content": "/**
  This CodeSandbox has been automatically generated using a cool tool.

  This generator does not follow relative imports beyond those that reference the
  module root, and as such, other relative imports may fail to load.

  If you are experiencing difficulty contact @noviny on twitter
*/
import React from 'react';
import ReactDOM from 'react-dom';
import Example from './example';

ReactDOM.render(
<Example />,
document.getElementById('root')
);",
    },
    "package.json": Object {
      "content": "{
  \\"name\\": \\"simple-example\\",
  \\"version\\": \\"0.0.0\\",
  \\"description\\": \\"A simple example deployed to CodeSandbox\\",
  \\"main\\": \\"index.js\\",
  \\"dependencies\\": {\\"b\\":\\"v1.0.0\\",\\"not-a-real-name\\":\\"0.0.1\\",\\"d\\":\\"1.1.0\\",\\"react\\":\\"latest\\",\\"react-dom\\":\\"latest\\"}
}",
    },
  },
  "parameters": "N4IgZglgNgpgziAXKCA7AJjAHgOgBYAuAtlEqAMYD2qBMNSIAPOhAG4AEE6AvADogAnSpQL8AfIwD0LVmJABfADQg0mXACsEyEFRp0CDSQCojvVO3YAVPBDjsAwpUwBlAIYYARpSzs8rux4wdOyuAK4ElESuBBDkrlBQAJ7sAOZ0MALRMOjsoXBoKSHsVJRQ7BGlOGZmFta2qemZEQLs6JTw7Kgi7GClUJQA7uwCMFDRbDCcRAAOlAIEATCJ1DkEeJRwk2vRwzBgGXTkW3gwNexETqGww8IEiiEYIXZwoeR49yInLSNjMayTEBmcwW51cyTArmg5Uo7H6rnQVVQZwAkmB2MtQiERuxsNMMhBDgVWhAwJByFcCMldARXOQCOwAAJdVhoZLUcoDCAEWgCMxGSRmQGzebsABKMFp9LAQiI7AA5CNJXKANyCoEi8WSgAiAHkALI9GXyxV0gC0bSIKrVwvpAFEsK4ZtdpZF5ThJNhHdNYFakahNXTdXqcCMMBkABRmRj2r3XSRiRRmNrkoj6HBpAi22CpmgAIUSyPQ4YVtzlAEozGXlQplJ6nTAcJoyDpqLR6IgVOr6a5Da65R4VVMbcVe7K5e7yHKayBprSANauNKNuDUZvU_QMYBnfioR2nJDsfj5eumuve_eJ8yHkD_AT5aj8RDXgAMOFfz_4l4s_EwcHIAggaYYgfA9-AAQXYY9zxxB161aGBvUoRJsmhBwnBgNxPG8T9txAKI0Efa9VGwZccKvH8ELoTBUHIAk4EIrcQA8R9-FYABGN83xwkAugIU1XFNRUoFNXdUxYkB3xwNjuPQcSOI4j8QETQQJTpcTfngUQlP4E0-ItdSsjgLT5DMeQFHkeQgA",
}
`;

exports[`getCSBData() simple parse 1`] = `
Object {
  "dependencies": Object {
    "b": "v1.0.0",
    "not-a-real-name": "0.0.1",
    "react": "latest",
    "react-dom": "latest",
  },
  "files": Object {
    "example.js": Object {
      "content": "import a from 'b'; import c from './c'",
    },
    "index.html": Object {
      "content": "<div id=\\"root\\"></div>",
    },
    "index.js": Object {
      "content": "/**
  This CodeSandbox has been automatically generated using a cool tool.

  This generator does not follow relative imports beyond those that reference the
  module root, and as such, other relative imports may fail to load.

  If you are experiencing difficulty contact @noviny on twitter
*/
import React from 'react';
import ReactDOM from 'react-dom';
import Example from './example';

ReactDOM.render(
<Example />,
document.getElementById('root')
);",
    },
    "package.json": Object {
      "content": "{
  \\"name\\": \\"simple-example\\",
  \\"version\\": \\"0.0.0\\",
  \\"description\\": \\"A simple example deployed to CodeSandbox\\",
  \\"main\\": \\"index.js\\",
  \\"dependencies\\": {\\"b\\":\\"v1.0.0\\",\\"not-a-real-name\\":\\"0.0.1\\",\\"react\\":\\"latest\\",\\"react-dom\\":\\"latest\\"}
}",
    },
  },
  "parameters": "N4IgZglgNgpgziAXKCA7AJjAHgOgBYAuAtlEqAMYD2qBMNSIAPOhAG4AEE6AvADogAnSpQL8AfIwD0LVmJABfADQg0mXACsEyEFRp0CDSQCojvVO3YAVPBDjsAwpUwBlAIYYARpSzs8rux4wdOyuAK4ElESuBBDkrlBQAJ7sAOZ0MALRMOjsoXBoKSHsVJRQ7BGlOGZmFta2qemZEQLs6JTw7Kgi7GClUJQA7uwCMFDRbDCcRAAOlAIEATCJ1DkEeJRwk2vRwzBgGXTkW3gwNexETqGww8IEiiEYIXZwoeR49yInLSNjMayTEBmcwW51cyTArmg5Uo7H6rnQVVQZwAkmB2MtQiERuxsNMMhBDgVWhAwJByFcCMldARXOQCOwAAJdVhoZLUcoDCAEWgCMxGSRmQGzebsABKMFp9LAQiI7AA5CNJXKANyCoEi8WSgAiAHkALI9GXyxV0gC0bSIKrVwvpAFEsK4ZtdpZF5ThJNhHdNYFakahNXTdXqcCMMBkABRmRj2r3XSRiRRmNrkoj6HBpAi22CpmgAIUSyPQ4YVtzlAEozGXlQplJ6nTAcJoyDpqLR6IgVOr6a5Da65R4VVMbcVe7K5e7yHKayBprSANauNKNuDUZvU_QMYBnfioR2nJDsfj5eumuve_eJ8yHkD_AT5aj8RDXgAMOFfz_4l4s_EwcHIAggaYYgfA9-AAQXYY9zxxB161aGBvUoRJsmhBwnBgNxPG8T9txAKI0Efa9VGwZccKvH8ELoTBUHIAk4EIrcQA8R9-FYABGN83xwkAugIU1XFNRUoFNXdUxYkB3xwNjuJNUQkH4X54DkxNBAlM0LXExS4Dk-QzHkBR5HkIA",
}
`;

exports[`parseDeps() dev and peer deps 1`] = `
Object {
  "deps": Object {
    "t": "latest",
    "x": "v1.0.0",
    "z": "v1.0.0",
  },
  "internalImports": Array [],
}
`;

exports[`parseDeps() multiline imports 1`] = `
Object {
  "deps": Object {
    "b": "v1.0.0",
    "d": "latest",
  },
  "internalImports": Array [],
}
`;

exports[`parseDeps() no spaces 1`] = `
Object {
  "deps": Object {
    "b": "v1.0.0",
  },
  "internalImports": Array [],
}
`;

exports[`parseDeps() relativeImport 1`] = `
Object {
  "deps": Object {},
  "internalImports": Array [
    Array [
      "import {a} from './c'",
      "./c",
    ],
  ],
}
`;

exports[`parseDeps() simple import 1`] = `
Object {
  "deps": Object {
    "b": "v1.0.0",
  },
  "internalImports": Array [],
}
`;

exports[`parseDeps() spread import 1`] = `
Object {
  "deps": Object {
    "b": "v1.0.0",
  },
  "internalImports": Array [],
}
`;

exports[`parseDeps() two imports 1`] = `
Object {
  "deps": Object {
    "b": "v1.0.0",
    "d": "latest",
  },
  "internalImports": Array [],
}
`;

exports[`parseDeps() two spread imports 1`] = `
Object {
  "deps": Object {
    "c": "latest",
  },
  "internalImports": Array [],
}
`;

exports[`parseDeps() two spread imports multiline 1`] = `
Object {
  "deps": Object {
    "c": "latest",
  },
  "internalImports": Array [],
}
`;

exports[`parseDeps() using regex pattern 1`] = `
Object {
  "deps": Object {},
  "internalImports": Array [
    Array [
      "import a from './c/somewhere'",
      "./c/somewhere",
    ],
    Array [
      "import b from './c/anywhere'",
      "./c/anywhere",
    ],
  ],
}
`;

exports[`replaceImport() dev and peer deps 1`] = `"import {a} from 't' import s from 'z' import y from 'x'"`;

exports[`replaceImport() multiline imports 1`] = `
"import a from 'zzz'
import c from 'd'"
`;

exports[`replaceImport() no spaces 1`] = `"import {a} from 'zzz'"`;

exports[`replaceImport() relativeImport 1`] = `"import {a} from './c'"`;

exports[`replaceImport() simple import 1`] = `"import a from 'zzz'"`;

exports[`replaceImport() spread import 1`] = `"import { a } from 'zzz'"`;

exports[`replaceImport() two imports 1`] = `"import a from 'zzz' import c from 'd'"`;

exports[`replaceImport() two spread imports 1`] = `"import { a, b } from 'c'"`;

exports[`replaceImport() two spread imports multiline 1`] = `
"import {
  a,
  b
} from 'c'"
`;

exports[`replaceImport() using regex pattern 1`] = `"import a from 'zzz' import b from 'zzz'"`;

exports[`replaceImports() replace imports 1`] = `"import a from 'z' import c from 'arg'"`;
